File: Seminar_Znakomstvo_Python/Seminar_5.Рекурсия и алгоритмы/task_1.py
# Последовательностью Фибоначчи называется последовательность чисел a0, a1 , ..., an, ..., 
# где a0 = 0, a1  = 1, ak = ak-1 + ak-2 (k > 1).

# Требуется найти N-е число Фибоначчи

# Input: 7
# Output: 21


def Fibon(num):
    if num <= 1:
        return 1
    return Fibon(num - 1) + Fibon(num - 2)


num = 5

print('Список фибоначчи')
for i in range(1, num+1):
    print(Fibon(i), end = ' ')
print()
print('Найден последний элемент под номером', num, '=', Fibon(num))

File: Seminar_Obrabotka/src/HW/HW_2/task_1/task_1.java
package HW.HW_2.task_1;

public class task_1 {
    static class IsFloat {
        /*
        Напишите программу, которая проверяет, является ли введенная текстовая строка числом с плавающей точкой (float).
        Программа пытается преобразовать введенную строку в число float, и если это успешно, она выводит полученное число.
        Если преобразование не удалось, программа выдаёт сообщение об ошибке и возвращает специальное значение Float.NaN,
        чтобы обозначить ошибку.
        На входе:
            '3.14'
            'Ivan'
        На выходе:
            3.14
         */
        public static float isFloat(String input) {
        // Введите свое решение ниже
            float num = 0;
            try {
                num = Float.parseFloat(input);
                return num;
            } catch (NumberFormatException | NullPointerException nfe) {
                System.out.println("Your input is not a float number. Please, try again.");
            }
            return Float.NaN;
        }
    }

    // Не удаляйте этот класс - он нужен для вывода результатов на экран и проверки
    public static class Printer {
        public static void main(String[] args) {
            String input;

    // При отправке кода на Выполнение, вы можете варьировать эти параметры
            if (args.length == 0) {
                input = "fff"; // По умолчанию используем "3.14", если аргумент не передан
            } else {
                input = args[0];
            }

            float result = IsFloat.isFloat(input);
            System.out.println(result);
        }
    }
}
